packages/react-components/src/trace-explorer/trace-explorer-opened-traces-widget.tsx

```diff
@@ -0,0 +1,322 @@
+import { OutputDescriptor } from 'tsp-typescript-client/lib/models/output-descriptor';
+import { signalManager, Signals } from '@trace-viewer/base/lib/signals/signal-manager';
+import ReactModal from 'react-modal';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCopy } from '@fortawesome/free-solid-svg-icons';
+import { AvailableAnalysesChangedSignalPayload } from '@trace-viewer/base/lib/signals/available-analyses-changed-signal-payload';
+import { OpenedTracesUpdatedSignalPayload } from '@trace-viewer/base/lib/signals/opened-traces-updated-signal-payload';
+import { ITspClientProvider } from '@trace-viewer/base/lib/tsp-client-provider';
+
+export interface ReactOpenTracesWidgetProps {
+    id: string,
+    title: string,
+    tspClientProvider: ITspClientProvider,
+    contextMenuRenderer?: (event: React.MouseEvent<HTMLDivElement>, experiment: Experiment) => void,
+    onDoubleClick?: (event: React.MouseEvent<HTMLDivElement>, experiment: Experiment) => void
+}
+
+export interface ReactOpenTracesWidgetState {
+    openedExperiments: Array<Experiment>
+}
+
+export class ReactOpenTracesWidget extends React.Component<ReactOpenTracesWidgetProps, ReactOpenTracesWidgetState> {
+    static LIST_MARGIN = 2;
+    static LINE_HEIGHT = 16;
+
+    protected forceUpdateKey = false;
+
+    sharingLink = '';
+    showShareDialog = false;
+    lastSelectedOutputIndex = -1;
+
+    private _selectedExperimentIndex = 0;
+    private _availableOutputDescriptors: Map<string, OutputDescriptor[]> = new Map();
+    private _selectedExperiment: Experiment | undefined;
+    private _experimentManager: ExperimentManager;
+
+    constructor(props: ReactOpenTracesWidgetProps) {
+        super(props);
+        signalManager().on(Signals.EXPERIMENT_OPENED, (experiment: Experiment) => this.onExperimentOpened(experiment));
+        signalManager().on(Signals.EXPERIMENT_CLOSED, (experiment: Experiment) => this.onExperimentClosed(experiment));
+        signalManager().on(Signals.TRACEVIEWERTAB_ACTIVATED, (experiment: Experiment) => this.onOpenedTracesWidgetActivated(experiment));
+
+        this._experimentManager = this.props.tspClientProvider.getExperimentManager();
+        this.props.tspClientProvider.addTspClientChangeListener(() => {
+            this._experimentManager = this.props.tspClientProvider.getExperimentManager();
+        });
+        this.state = { openedExperiments: [] };
+    }
+
+    componentDidMount(): void {
+        this.initialize();
+    }
+
+    componentWillUnmount(): void {
+        signalManager().off(Signals.EXPERIMENT_OPENED, (experiment: Experiment) => this.onExperimentOpened(experiment));
+        signalManager().off(Signals.EXPERIMENT_CLOSED, (experiment: Experiment) => this.onExperimentClosed(experiment));
+        signalManager().off(Signals.TRACEVIEWERTAB_ACTIVATED, (experiment: Experiment) => this.onOpenedTracesWidgetActivated(experiment));
```

Comment:
The listener does not get removed because it is not the same instance being passed that was added earlier. (3x)